daml-foundations/daml-ghc/tests/No_concat_clash.daml

```diff
@@ -0,0 +1,22 @@
+-- Copyright (c) 2019, Digital Asset (Switzerland) GmbH and/or its
+-- affiliates.  All rights reserved.
+
+-- This is a compile-time test.
+
+{-# LANGUAGE NoImplicitPrelude #-}
+
+-- Show that if the names of `Prelude` are not in scope, then there
+-- should be no ambiguity about `concat` introduced by `DA.Foldable`.
+
+daml 1.2
+module No_concat_clash where
+
+import Prelude qualified
+import DA.Foldable  -- for `concat`.
+
+template T
+  with
+    ps : [Prelude.Party]
+  where
+    signatory ps
```

Comment:
It would be good to have two `signatory` clauses to make sure we actually produce a reference to `concat` during desugaring. I can envision future optimizations which omit the `concat` when there's only one `signatory` clause.

Reply:
IIRC, we do validation checks so in fact at this time, two `signatory` clauses would be a hard error.